### PR TITLE
Fix py2/py3 reqs

### PR DESCRIPTION
**Pull request by MartinBasti**
# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

**Review by lkolacek — APPROVED** [review]
LGTM :+1:

**Comment by MartinBasti** [comment]
@lkolacek updated

File: tests/plugins/test_resolve_composes.py [hunk 1]
```diff
@@ -1363,7 +1364,10 @@ def run_plugin_with_args(self, workflow, plugin_args=None,
         if expect_error:
             with pytest.raises(PluginFailedException) as exc_info:
                 runner.run()
-            assert expect_error in str(exc_info.value)
+            if hasattr(expect_error, 'search'):  # py2/3 compat, different re.Compile objects
```
Comment (tkdchen) [inline tests/plugins/test_resolve_composes.py:1367]: How about use `six.PY2` or `six.PY3`? No comment is required and the code is much straightforward.

Comment (MartinBasti) [reply tests/plugins/test_resolve_composes.py:1367]: Can you provide exaple? I don't think that using six makes code more straightforward in this case

Comment (ben-alkov) [reply tests/plugins/test_resolve_composes.py:1367]: Actually, I'd like to see the comment explain *which* Python re.Compile object has  `search` (so I don't have to look it up :p)

**Comment by MartinBasti** [comment]
> Actually, I'd like to see the comment explain which Python re.Compile object has search (so I don't have to look it up :p)

```
>>> r = re.compile('test')
# py2
>>> type(r)
<type '_sre.SRE_Pattern'>

# py3
>>> type(r)
<class 're.Pattern'>
```

I'm not sure if I understand what you mean. It differs between python versions. In py2 it's not even object

**Comment by ben-alkov** [comment]
> I'm not sure if I understand what you mean. It differs between python versions. In py2 it's not even object

IOW, maybe the comment could be something like

```python
if hasattr(expect_error, 'search'):  # For Py 3; Py 2's re isn't an object
    assert expect_error.search(str(exc_info.value))
else:
    assert expect_error in str(exc_info.value) # Py 2
```
or whatever you're comfortable with :)

**Comment by MartinBasti** [comment]
Oh, it's misunderstanding. if-else is there because I want to support both string and regexp.

I added comment about py2/3 compatibility, because I cannot use `isinstance(expect_error, re.Pattern)` in py2

**Comment by MartinBasti** [comment]
No release notes, no functional change

**Pull request merged**